src/lib/alloc.c

```diff
@@ -457,6 +457,20 @@ static void free_block(void *ptr)
 	bool heap_is_full;
 
 	heap = get_heap_from_ptr(ptr);
+
+/* for multi-core support, the caller might ask for freeing buffer
+ * with pointer to uncached buffer while the allocation is from cached
+ * buffer, try to handle the freeing of this kind of buffer.
+ */
+#if CONFIG_CORE_COUNT > 1
+	if (!heap && is_uncached(ptr)) {
+		ptr = uncache_to_cache(ptr);
+		tr_info(&mem_tr, "free_block(): uncache buffer, try the cached buffer %p",
+			ptr);
+
+		heap = get_heap_from_ptr(ptr);
+	}
+#endif
```

Comment:
I do hope this stays in the Xtos variant only, or that it's a temporary solution waiting for an improved one further on. If either is the case, approved.

Reply:
Lets just keep it simple. Whats's wrong with always doing
```c
ptr = uncache_to_cache(ptr); // no effect if already cached or is platform does not use it.
heap = get_heap_from_ptr(ptr);
```

Reply:
> Lets just keep it simple. Whats's wrong with always doing
> 
> ```c
> ptr = uncache_to_cache(ptr); // no effect if already cached or is platform does not use it.
> heap = get_heap_from_ptr(ptr);
> ```

For the conditional compilation, it is needed as for single core platforms (e.g. byt, IMX) there is no "uncache_to_cache()" definition so without it you will see errors like below:
```
/home/keyon/hdd/github/keyonjie/sof/src/lib/alloc.c: In function ‘free_block’:
/home/keyon/hdd/github/keyonjie/sof/src/lib/alloc.c:466: warning: implicit declaration of function ‘is_uncached’
/home/keyon/hdd/github/keyonjie/sof/src/lib/alloc.c:467: warning: implicit declaration of function ‘uncache_to_cache’
```

For the trace_info(), I just want to print out this information as this actually should not be a usual case.

Reply:
Lets just define uncache_to_cache() for all platforms

Reply:
> Lets just define uncache_to_cache() for all platforms

Okay, let me change.